st2api/st2api/controllers/v1/pack_configs.py

```diff
@@ -63,3 +77,28 @@ def get_one(self, pack_ref):
         """
         # TODO: Make sure secret values are masked
         return self._get_one_by_pack_ref(pack_ref=pack_ref)
+
+    @request_user_has_permission(permission_type=PermissionType.PACK_CREATE)
+    @jsexpose(body_cls=ConfigUpdateRequestAPI, arg_types=[str])
+    def put(self, pack_uninstall_request, pack_ref):
+        """
+            Create a new config for the action.
+
+            Handles requests:
+                POST /configs/<pack_ref>
+        """
+
+        try:
+            config_api = ConfigAPI(pack=pack_ref, values=vars(pack_uninstall_request))
+            config_api.validate(validate_against_schema=True)
+        except jsonschema.ValidationError as e:
+            raise ValueValidationException(str(e))
+
+        config_content = yaml.safe_dump(config_api.values, default_flow_style=False)
+
+        configs_path = os.path.join(cfg.CONF.system.base_path, 'configs/')
```

Comment:
Also - I assume it's still the same as issue we have in other places where we write content to the disk from inside the API.

So probably we don't have a good solution for it (yet), but adding more things which write to disk directly just increases the scope of the problem (which I guess is fine for now as long as we tell users how to sync the changes and work on multi node systems).